kotlin-spark-api/2.4/src/main/kotlin/org/jetbrains/kotlinx/spark/api/UDFRegister.kt

```diff
@@ -0,0 +1,803 @@
+@file:Suppress("NOTHING_TO_INLINE", "DuplicatedCode", "MemberVisibilityCanBePrivate")
```

Comment:
If this file is generated (I really hope it is) please provide us with generating code in comments. Thanks!

Reply:
> Sorry for not answering for a long time, was busy with another project. It's very good PR, but needs several iprovements.

No problem, and i thought so (especially with the improvements). This was only some kind of strange brainwave of mine and i hacked something together.

I will provide the generator-code after some clean-up, i only hacked it down for my own project and thought you might like/need it. :D 
If you have further suggestions please let me know, i'll try to add them to the implementation.

(sorry that i can't do it today, but it's 00:30 in germany and i have a meeting in the morning.)

Reply:
There is no hurry at all, thank you for your effort.
How do you think if there is any way to support other classes natively supported by Spark?

Reply:
In my fork is a branch for calling the scala-java-conversion-wrappers (i didnt call for a merge request, because i'm not really satisfied with the tests for it. i have problems testing the mutable scala types for changes, if you could help me with that [or give me some tipps how to write them] i  would be thankfull).

I think we can achieve some kind of "auto-conversion" by using reified and calling these wrapper-functions. (at least for the function-wrappers)

I wrote this text on my smartphone, please excuse the typing error.

Reply:
I think if you would file a PR we at least could think on how to test something :)

Reply:
I'll take care for the code around christmas. Right now i don't have enough time to work on my PRs besides my current tasks.
I also commited the PR for the Wrappers (see https://github.com/JetBrains/kotlin-spark-api/pull/72)

Thank you for your feedback in advance! 😄